packages/react/src/context/ComlinkTokenRefresh.tsx

```diff
@@ -0,0 +1,190 @@
+  const authState = useAuthState()
+
+  const clearRefreshTimeout = useCallback(() => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current)
+      timeoutRef.current = null
+    }
+  }, [])
+
+  const windowConnection = useWindowConnection<SdkParentComlinkMessage, SdkChildComlinkMessage>({
+    name: SDK_NODE_NAME,
+    connectTo: SDK_CHANNEL_NAME,
+  })
+
+  const requestNewToken = useCallback(async () => {
+    if (isTokenRefreshInProgress.current) {
+      return
+    }
+
+    isTokenRefreshInProgress.current = true
+    clearRefreshTimeout()
+
+    timeoutRef.current = setTimeout(() => {
+      if (isTokenRefreshInProgress.current) {
+        isTokenRefreshInProgress.current = false
+      }
+      timeoutRef.current = null
+    }, DEFAULT_RESPONSE_TIMEOUT)
+
+    try {
+      const res = await windowConnection.fetch<{token: string | null; error?: string}>(
+        'dashboard/v1/auth/tokens/create',
+      )
+      clearRefreshTimeout()
+
+      if (res.token) {
+        setAuthToken(instance, res.token)
+      }
+      isTokenRefreshInProgress.current = false
+    } catch {
+      isTokenRefreshInProgress.current = false
+      clearRefreshTimeout()
+    }
+  }, [windowConnection, clearRefreshTimeout, instance])
+
+  const contextValue = useMemo(
+    () => ({
+      requestNewToken,
+      isTokenRefreshInProgress,
+    }),
+    [requestNewToken],
+  )
+
+  useEffect(() => {
+    return () => {
+      clearRefreshTimeout()
+    }
+  }, [clearRefreshTimeout])
+
+  useEffect(() => {
+    const has401Error =
+      authState.type === AuthStateType.ERROR &&
+      authState.error &&
+      (authState.error as ClientError)?.statusCode === 401 &&
+      !isTokenRefreshInProgress.current &&
+      processed401ErrorRef.current !== authState.error
+
+    const isLoggedOut =
+      authState.type === AuthStateType.LOGGED_OUT && !isTokenRefreshInProgress.current
+
+    if (has401Error || isLoggedOut) {
+      processed401ErrorRef.current =
+        authState.type === AuthStateType.ERROR ? authState.error : undefined
+      requestNewToken()
+    } else if (
+      authState.type !== AuthStateType.ERROR ||
+      processed401ErrorRef.current !==
+        (authState.type === AuthStateType.ERROR ? authState.error : undefined)
+    ) {
+      processed401ErrorRef.current = null
+    }
+  }, [authState, requestNewToken])
+
+  return (
+    <ComlinkTokenRefreshContext.Provider value={contextValue}>
+      {children}
+    </ComlinkTokenRefreshContext.Provider>
+  )
+}
+
+/**
+ * Component that provides a no-op token refresh outside of dashboard
+ */
+function NoOpTokenRefresh({children}: PropsWithChildren) {
+  const contextValue = useMemo(
+    () => ({
+      requestNewToken: () => {},
+      isTokenRefreshInProgress: {current: false},
+    }),
+    [],
+  )
+
+  return (
+    <ComlinkTokenRefreshContext.Provider value={contextValue}>
+      {children}
+    </ComlinkTokenRefreshContext.Provider>
+  )
+}
```

Comment:
i don't think you need a no-op provider. i think you could omit the hook entirely with the current implementation and only continue to conditionally wrap with the DashboardTokenRefresh provider component and it should still work as expected